src/chatgpt.py

```diff
@@ -487,14 +576,25 @@ def main(
     config["json_mode"] = json_mode
 
     copyable_blocks = {} if config["easy_copy"] else None
-
+    if config["supplier"] == "azure":
+        model = config["AZURE_OPENAI_DEPLOYMENT_NAME"]
+    elif config["supplier"] == "openai":
+        model = config["OPENAI_MODEL"]
```

Comment:
More switching on the `supplier` - makes me think that having classes as mentioned before will be nice.

Reply:
Your consideration is correct. In general all the code is becoming a little bit messy after many addition and it needs a general refactoring. I will probably do it during Christmas vacation.